packages/dev/core/src/Engines/engine.ts

```diff
@@ -421,7 +421,19 @@ export class Engine extends ThinEngine {
     /**
      * Gets a boolean indicating if the pointer is currently locked
      */
-    public isPointerLock = false;
+    public get isPointerLock() {
+        // Return either manual pointerlock state or canvas pointerlock state
+        return this._isPointerLock || (this._isCanvasPresent && document.pointerLockElement === this._renderingCanvas);
+    }
+
+    public set isPointerLock(value: boolean) {
+        this._isPointerLock = value;
+    }
+
+    // Store manual pointerlock state, if specified
+    private _isPointerLock = false;
+    // Check for document and canvas before using pointerlock
+    private _isCanvasPresent = false;

```

Comment:
It seems like we only check document?

```suggestion
    private _isDocumentPresent = false;
```

Reply:
Never mind, this only gets set if there is a canvas.